examples/hello/hello_main.c

```diff
@@ -50,6 +54,12 @@
 
 int main(int argc, FAR char *argv[])
 {
+#ifdef CONFIG_SMP
+  uint32_t cpu = up_cpu_index();
```

Comment:
Neither arevproper posix interfaces and neither show be called or implemted in you space.  This must be reverted.

It will also break any SMP PROTECTED or kernel mode build.

You could make an exteneion to prctl().  The at least uses a Linux-like interface and will not break other builds.

Reply:
@patacongo 
thanks for pointing out the protected and kernel builds I will take then into consideration

Reply:
I would revert sched_getcpu() as well. All OS application interfaces must be defined at opengroup.org or use an well established hook like ioctl(), prctl(), boarctl(), and maybe sysctl().

Reply:
Many of the things in boardctl() really should be moved to a new sysctl() what handles all system-level setting.  All task-level setting (including CPU) should be provided by prctl() (process control).  Only board level stuff should be included in boardctl().